honeybee/search.py

```diff
@@ -0,0 +1,41 @@
+"""Collection of methods for searching for keywords and filtering lists by keywords."""
+
+
+def filter_array_by_keywords(array, keywords):
+    """Filter an array of strings to get only those containing the given keywords.
+
+    This method is case insensitive, allowing the searching of keywords across
+    different cases of letters. It will also automatically parse any strings of
+    multiple keywords (spearated by spaces) into separate keywords for searching.
+
+    Args:
+        array: An array of strings which will be filtered to get only those containing
+            the given keywords.
+        keywords: An array of strings representing keywords.
+    """
+    # split any keywords separated by spaces
+    split_keywords = []
+    for kw in keywords:
+        split_keywords.extend(kw.strip().upper().split(' '))
```

Comment:
This is tricky! What if someone does want to search for the name with space inside? I think we should either give an option for split or let user split the keywords before passing them to this function as an input.

In any case, I would do it slightly different than this to handle cases with more than a single space with no issues (e.g. 'this     keyword'.

```suggestion
        split_keywords.extend(kw.upper().split())
```

Also why don't you use list comprehension in this case?

Reply:
We never had this case in honeybee-legacy (this splitting method was always the preferred one).  Still, I will add an option to not split it for the cases people might need the space in the search.

List comprehension is always a fine substitute for looping and appending be we need to use extend here. I can do a for loop within a for loop using list comprehension here but that is usually where I draw the line and just say that looping and extending is just as elegant.  I'll switch it to list comprehension since I see that you have a stronger preference for it.

Reply:
Now that you mentioned it, I just thought of one of the new cases where I want to apply this method and I realized it's one where including the space is important. So I wholeheartedly agree that we should expose it as an option.